File: 2026/April/LC-3740-Minimum-Distance-Between-Three-Equal-Elements-I/solution.cpp
// Approach 1:
class Solution {
    public:
        int minimumDistance(vector<int>& nums) {
            int n=nums.size(),mini=1e9;
    
            for(int i=0;i<n-2;i++){
                for(int j=i+1;j<n-1;j++){
                    for(int k=j+1;k<n;k++){
                        if(nums[i]==nums[j] && nums[j]==nums[k]){
                            int dist=abs(i-j)+abs(j-k)+abs(k-i);
                            mini=min(mini,dist);
                        }
                    }
                }
            }
    
            return mini==1e9?-1:mini;
        }
    };

// Approach 2:
class Solution {
    public:
        int minimumDistance(vector<int>& nums) {
            int min_diff = 201;
            unordered_map<int, vector<int>> indices;
            for (int i = 0; i < nums.size(); i++) {
                int n = indices[nums[i]].size();
                if (n >= 2) {
                    min_diff = min(min_diff, i - indices[nums[i]][n - 2]);
                }
                indices[nums[i]].push_back(i);
            }
            return (min_diff != 201) ? min_diff * 2 : -1;
        }
    };

File: 2026/April/LC-3740-Minimum-Distance-Between-Three-Equal-Elements-I/solution.py
# Approach 1:
from typing import List
from collections import defaultdict
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        n = len(nums)
        if n <= 2 or len(set(nums)) == n:
            return -1
        ht = defaultdict(list)
        for i, v in enumerate(nums):
            ht[v].append(i)
        mnd = 1000
        for x, f in ht.items():
            l = len(f)
            if l >= 3:
                for i in range(l - 2):
                    for j in range(i + 1, l - 1):
                        for k in range(j + 1, l):
                            mnd = min(
                                mnd,
                                abs(f[i] - f[j]) + abs(f[j] - f[k]) + abs(f[k] - f[i]),
                            )
        return mnd if mnd != 1000 else -1

# Approach 2:
from typing import List
class Solution:
    def minimumDistance(self, nums: List[int]) -> int:
        n = len(nums)
        if n <= 2 or len(set(nums)) == n:
            return -1
        mnd = 1000
        for i in range(n-2):
            for j in range(i + 1, n-1):
                for k in range(j + 1, n):
                    if nums[i] == nums[j] and nums[j] == nums[k]:
                        dist: int = abs(i - j) + abs(j - k) + abs(i - k)
                        mnd: int = min(mnd, dist)
        return mnd if mnd != 1000 else -1

# Approach 3:
from collections import defaultdict

class Solution:
    def minimumDistance(self, nums: list[int]) -> int:
        prev_indexes: defaultdict[int, list[int]] = defaultdict(list)
        result: int = 10 ** 10
        for pos, num in enumerate(nums):
            if len(prev_indexes[num]) >= 2:                
                result = min(result, pos - prev_indexes[num][-2])                
            prev_indexes[num].append(pos)
        if result == 10 ** 10:
            return -1
        return result * 2